lib/Db/Row2Mapper.php

```diff
@@ -309,46 +309,84 @@ private function getFilterExpression(IQueryBuilder $qb, Column $column, string $
 		$paramType = $this->getColumnDbParamType($column);
 		$value = $this->formatValue($column, $value, 'in');
 
+		// We try to match the requested value against the default before building the query
+		// so we know if we shall include rows that have no entry in the column_TYPE tables upfront
+		$includeDefault = false;
+		$defaultValue = $this->getFormattedDefaultValue($column);
+
 		$qb2 = $this->db->getQueryBuilder();
-		$qb2->select('row_id');
-		$qb2->where($qb->expr()->eq('column_id', $qb->createNamedParameter($column->getId()), IQueryBuilder::PARAM_INT));
-		$qb2->from('tables_row_cells_' . $column->getType());
+		$qb2->selectAlias('sl.id', 'row_id')
+			->from('tables_row_sleeves', 'sl')
+			->leftJoin('sl', 'tables_row_cells_' . $column->getType(), 'v', $qb->expr()->andX(
+				$qb->expr()->eq('sl.id', 'v.row_id'),
+				$qb->expr()->eq('v.column_id', $qb->createNamedParameter($column->getId(), IQueryBuilder::PARAM_INT)),
+				$qb->expr()->eq('sl.table_id', $qb->createNamedParameter($column->getTableId(), IQueryBuilder::PARAM_INT))
```

Comment:
Being a left join and since it is not establishing a condition between `sl` and `v`, this is rather to be understood as part of the general where clause, no?

Reply:
Yes, good catch for the sl.table_id, the v.column_id we need to also include NULL values on the join where the row_id does not match

Reply:
Adjusted